src/bp/nlu-core/intents/exact-intent-classifier.ts

```diff
@@ -77,25 +83,33 @@ export class ExactIntenClassifier implements IntentClassifier {
       const model: Model = await validate(raw, modelSchema)
       this.model = model
     } catch (err) {
-      throw new ModelLoadingError(ExactIntenClassifier._name, err)
+      throw new ModelLoadingError(ExactIntenClassifier._displayName, err)
     }
   }
 
-  async predict(utterance: Utterance): Promise<IntentPredictions> {
+  async predict(utterance: Utterance): Promise<NoneableIntentPredictions> {
     if (!this.model) {
-      throw new Error(`${ExactIntenClassifier._name} must be trained before calling predict.`)
+      throw new Error(`${ExactIntenClassifier._displayName} must be trained before calling predict.`)
     }
 
+    const { _name: extractor } = ExactIntenClassifier
+
     const { exact_match_index, intents: intentNames } = this.model
+
     const exactPred = this._findExactIntent(exact_match_index, utterance)
 
     if (exactPred) {
+      const oneHot = intentNames.map(name => ({ name, confidence: name === exactPred ? 1 : 0, extractor }))
       return {
-        intents: [{ name: exactPred, confidence: 1, extractor: 'exact-matcher' }]
+        oos: 0,
+        intents: oneHot
       }
     }
+
+    const zeros = intentNames.map(name => ({ name, confidence: 0, extractor }))
     return {
-      intents: []
+      oos: 1,
```

Comment:
Will that bias the election step ? IOW will this potentially make the election step choose OOS : 1 thus more likely to elect none ?

Reply:
Nope, only a way for exact-intent-classifier to say if there's a pred or not. It makes the sum of confidences 1 so it's a true classifier.